src/source/Response.c

```diff
@@ -180,6 +180,35 @@ STATUS initializeCurlSession(PRequestInfo pRequestInfo, PCallInfo pCallInfo, CUR
         curl_easy_setopt(pCurl, CURLOPT_SSLVERSION, CURL_SSLVERSION_TLSv1_2);
     }
 
+#if defined(AWS_KVS_IPV4_ONLY)
+    curl_easy_setopt(pCurl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
+    DLOGD("Curl resolving IPV4 through AWS_KVS_IPV4_ONLY");
+#elif defined(AWS_KVS_IPV6_ONLY)
+    curl_easy_setopt(pCurl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V6);
+    DLOGD("Curl resolving IPV6 through AWS_KVS_IPV6_ONLY");
+#elif defined(AWS_KVS_DUAL_STACK_ONLY)
+    curl_easy_setopt(pCurl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_WHATEVER);
+    DLOGD("Curl resolving dual stack through AWS_KVS_DUAL_STACK_ONLY");
+#else
+    if (!IS_NULL_OR_EMPTY_STRING(GETENV(KVS_CURL_IPRESOLVE_V4_ENV_VAR))) {
+        curl_easy_setopt(pCurl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
+        DLOGD("Curl resolving IPV4 through environment variable");
+    } else if (!IS_NULL_OR_EMPTY_STRING(GETENV(KVS_CURL_IPRESOLVE_V6_ENV_VAR))) {
+        curl_easy_setopt(pCurl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V6);
+        DLOGD("Curl resolving IPV6 through environment variable");
+    } else {
+        curl_easy_setopt(pCurl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_WHATEVER);
+        DLOGD("Curl resolving dual stack by default");
+    }
+#endif
+
+#ifndef NDEBUG
+    // Only available in debug build
+    if (!IS_NULL_OR_EMPTY_STRING(GETENV(KVS_CURL_DEBUG_ENV_VAR))) {
+        curl_easy_setopt(pCurl, CURLOPT_VERBOSE, 1L);

```

Comment:
Can you please add a comment here describing the intentions one would have for using this mode

Reply:
Sure. This would add curl ultra-debug logs https://ec.haxx.se/usingcurl/verbose/index.html. The relevant part here is that it will also print out the IP address that it resolved and connected to.